src/NewPost.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { data } from "./data";
+
+export class NewPost extends Component {
+  state = {
+    id: 0,
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: ""
+  };
+
+  createDateOnSubmit = () => {
+    const date = new Date();
+    let year = date.getFullYear();
+    let month = date.getMonth();
+    let day = date.getDate();
+    const dateNow = `${year}-${month}-${day}`;
+    return dateNow;
+  };
+
+  createUniqueIdOnSubmit = () => {
+    const id = data.length;
+    return id;
+  };
+
+  convertTagsToArray = tags => {
+    let resultArray = [];
```

Comment:
you don't need to start with this assignment. you're always going to do `tags.split(splitOn)`, so you can just start with `const resultArray = tags.split(splitOn)`. you can also use `const` instead of `let` with this array for the same reason as with `blogPost` below

Reply:
I changed let to const and got an error message saying "resultArray is a const" and red squiggly lines under remaining "resultArray" The error message went away as did the red squiggles when I changed const back to let.